src/Microsoft.AspNetCore.Blazor.Server/Circuits/DefaultCircuitFactory.cs

```diff
@@ -2,32 +2,36 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using Microsoft.AspNetCore.Blazor.Browser.Rendering;
-using Microsoft.AspNetCore.Blazor.Builder;
 using Microsoft.AspNetCore.Blazor.Rendering;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.SignalR;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.AspNetCore.Blazor.Server.Circuits
 {
     internal class DefaultCircuitFactory : CircuitFactory
     {
         private readonly IServiceScopeFactory _scopeFactory;
+        private readonly DefaultCircuitFactoryOptions _options;
 
-        public DefaultCircuitFactory(IServiceScopeFactory scopeFactory)
+        public DefaultCircuitFactory(
+            IServiceScopeFactory scopeFactory,
+            IOptions<DefaultCircuitFactoryOptions> options)
         {
-            _scopeFactory = scopeFactory ?? throw new ArgumentNullException(nameof(scopeFactory));
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
 
-            StartupActions = new Dictionary<PathString, Action<IBlazorApplicationBuilder>>();
+            _scopeFactory = scopeFactory ?? throw new ArgumentNullException(nameof(scopeFactory));
+            _options = options.Value;
         }
 
-        public Dictionary<PathString, Action<IBlazorApplicationBuilder>> StartupActions { get; }
-
         public override CircuitHost CreateCircuitHost(HttpContext httpContext, IClientProxy client)
         {
-            if (!StartupActions.TryGetValue(httpContext.Request.Path, out var config))
+            if (!_options.StartupActions.TryGetValue(httpContext.Request.Path, out var config))
```

Comment:
I moved the `StartupActions` onto a new config class so we can use the DI system's `Configure` callbacks to register multiple startup actions (one for each `AddServerSideBlazor<TClient>` call). For now you can only call that once, because there's no way to specify distinct paths, but in the future we could have `AddServerSideBlazor` take an optional "path" arg.